plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,941 @@
+		if dsrs.setting.state == connected {
+			dsrs.createRequests(ctx, c)
+		}
+		if dsrs.setting.state == created {
+			// Launch 2 goroutines: one for reception of telemetry data,
+			//  and one for reception of event notifications
+			var waitgroup sync.WaitGroup
+
+			if len(dsrs.Subscriptions) > 0 {
+				waitgroup.Add(1)
+				go func() {
+					dsrs.receiveTelemetry(ctx, c)
+					waitgroup.Done()
+				}()
+			}
+
+			if len(dsrs.Notifications) > 0 {
+				waitgroup.Add(1)
+				go func() {
+					dsrs.receiveNotifications(ctx, c)
+					waitgroup.Done()
+				}()
+			}
+
+			waitgroup.Wait()
+		}
+
+		select {
+		case <-ctx.Done():
+		case <-time.After(time.Duration(c.Redial)):
+		}
+	}
+}
+
+// createRequests establishes dynamic dial-in subscriptions over NETCONF
+func (dsrs *dialinSubscriptionRequestsService) createRequests(
+	ctx context.Context, c *CiscoTelemetryNETCONF) error {
+	var err error
+
+	if len(dsrs.Subscriptions) > 0 {
+		// Initialization of subscriptions
+		subscriptions := make([]*netconfSubscriptionRequest, len(dsrs.Subscriptions))
+		for i, subscription := range dsrs.Subscriptions {
+			if subscription.Period != 0 {
+				if time.Duration(subscription.Period) >= 0 {
+					// Check type of subscription
+					switch subscription.UpdateTrigger {
+					case "periodic":
+						if time.Duration(subscription.Period) > 0 {
+							subscriptions[i] = &netconfSubscriptionRequest{
+								YangPush:    "urn:ietf:params:xml:ns:yang:ietf-yang-push",
+								Stream:      "yp:yang-push",
+								XPathFilter: subscription.XPathFilter,
+								Period: uint64(time.Duration(subscription.Period).Nanoseconds() /
+									(int64(time.Millisecond) * 10)),
+							}
+						} else {
+							err = fmt.Errorf(
+								"failed to create telemetry subscription: "+
+									"time period for subscription %d has to be "+
+									"strictly positive but is %d",
+								i+1,
+								time.Duration(subscription.Period),
+							)
+							c.acc.AddError(err)
+							continue
+						}
+					case "on-change":
+						p := new(uint64)
+						*p = uint64(time.Duration(subscription.Period).Nanoseconds() /
+							(int64(time.Millisecond) * 10))
+						// on-change
+						subscriptions[i] = &netconfSubscriptionRequest{
+							YangPush:        "urn:ietf:params:xml:ns:yang:ietf-yang-push",
+							Stream:          "yp:yang-push",
+							XPathFilter:     subscription.XPathFilter,
+							DampeningPeriod: p,
+						}
+					default:
+						err = fmt.Errorf(
+							"failed to create telemetry subscription: " +
+								"bad / missing field update_trigger " +
+								"(options are: periodic, on-change)",
+						)
+						c.acc.AddError(err)
+						continue
+					}
+
+					// Create dynamic subscription over NETCONF
+					log.Printf(
+						"%s: establishing subscription %s %s %s key(s)=%v...",
+						pluginName, subscription.XPathFilter,
+						subscription.UpdateTrigger, time.Duration(subscription.Period),
+						subscription.Keys,
+					)
```

Comment:
Please use Telegraf internal logger and use correct severity (Debug, Info, Warning, Critical, ..)

Reply:
Addressed by @NetworkGuy-0x06

Reply:
This seems not to be addressed yet, I still see `log.Printf` usages all over the place.